plugin/src/main/scala/io/github/davidgregory084/ScalacOptions.scala

```diff
@@ -653,6 +653,39 @@ private[davidgregory084] trait ScalacOptions {
     warnError
   )
 
+  /** Verbose options (-V)
+    */
+  def verboseOption(name: String, isSupported: ScalaVersion => Boolean = _ => true) =
+    ScalacOption(s"-V$name", isSupported)
+
+  /** Print dependent missing implicits.
+    */
+  val verboseImplicits =
+    verboseOption("implicits", _.isBetween(V2_13_0, V3_0_0))
+
+  /** Print found/required error messages as colored diffs.
+    */
+  val verboseTypeDiffs =
+    verboseOption("type-diffs", _.isBetween(V2_13_6, V3_0_0))
+
+  /** Explain type errors in more detail.
+    */
+  val explainTypes =
+    ScalacOption("-explaintypes", _ < V3_0_0)
+
+  /** Explain errors in more detail.
+    */
+  val explain = ScalacOption("-explain", _ >= V3_0_0)
```

Comment:
The [dotty source code](https://github.com/lampepfl/dotty/blob/main/compiler/src/dotty/tools/dotc/config/ScalaSettings.scala#L118) says that `-explain-types` is deprecated in favour of `-explain`, which solves our `explaintypes2`/`explaintypes3` naming problem.